### PR TITLE
Fixed bugs and add improvements for Newsletter Queue

### LINKED ISSUE #36612: [Issue] Fixed bugs and add improvements for Newsletter Queue
This issue is automatically created based on existing pull request: magento/magento2#33849: Fixed bugs and add improvements for Newsletter Queue

---------

### Description 
This PR fixes bugs and add improvements  for  Newsletter Queue functionality

### Preconditions
1. For correct testing you should have more then 20 subscribers in `newsletter_subscriber` table
2. Create new **Newsletter Templates** in **Marketing->Newsletter Templates** 
3. Create new **Newsletter Queue** 

https://user-images.githubusercontent.com/13517181/129901984-3874092d-06de-4142-9a2d-f350befdb1fc.mov

#### Actual result
1. Run `bin/magento cron:run --group=default` ,` newsletter_send_all` cron job should run **Newsletter Queue**
2. You will see that **20 emails were sent** but we still have **Not Sent** status 

Screenshot 2021-08-18 at 16 37 28

3. I don't see any actions except **Preview**

Screenshot 2021-08-18 at 16 44 20

#### Result after fix 
Was fixed by appliing next changes https://github.com/magento/magento2/pull/33849/files#diff-1d2d0f2340ff4fc3e0907e98b2ccd1a21a35c777c503907bc3c12f0223eaeb8eR254

Correct status was set, i can see **Pause**, **Cancel** , **Preview**  actions for queue

Screenshot 2021-08-18 at 16 48 58

## Issue case 2
1. After **Issue case 1** was fixed next issue was reproduced  
#### Actual result

All data were removed from Newsletter Queue after clicking **Save and Resume** button

https://user-images.githubusercontent.com/13517181/129911385-51cd3ae5-a676-4649-90f7-0176550e2f92.mov

Screenshot 2021-08-18 at 16 59 08

#### Result after fix 
Was fixed by appliing next changes  https://github.com/magento/magento2/pull/33849/files#diff-f4e00fe342edaab019fbb62abfa0c2b8f5d7a2f179bd5880e4c217d57bbb7883R232 

Data was saved correctly  all data present 

https://user-images.githubusercontent.com/13517181/129914043-f9a78778-f169-481c-b907-1b6a7dad6057.mov

#### Actual result
1. **Save and Resume** button always display  even while creating new  **Newsletter Queue**  

Screenshot 2021-08-18 at 17 04 30

#### Result after fix 
Was fixed by appliing next changes  https://github.com/magento/magento2/pull/33849/files#diff-d337328a9078b6f37102b714ffa5460b365c7ede7b97ca644bf2f0b8fe15b0dcR138

1. **Save and Resume** button displays  only if  **Newsletter Queue**  has status **Pause** 

https://user-images.githubusercontent.com/13517181/129914610-f2be7644-a867-40f8-af93-0424de400b15.mov

### PR DESCRIPTION
**Pull request by Usik2203**
### Description 
This PR fixes bugs and add improvements  for  Newsletter Queue functionality

### Preconditions
1. For correct testing you should have more then 20 subscribers in `newsletter_subscriber` table
2. Create new **Newsletter Templates** in **Marketing->Newsletter Templates** 
3. Create new **Newsletter Queue** 

https://user-images.githubusercontent.com/13517181/129901984-3874092d-06de-4142-9a2d-f350befdb1fc.mov

#### Actual result
1. Run `bin/magento cron:run --group=default` ,` newsletter_send_all` cron job should run **Newsletter Queue**
2. You will see that **20 emails were sent** but we still have **Not Sent** status 

Screenshot 2021-08-18 at 16 37 28

3. I don't see any actions except **Preview**

Screenshot 2021-08-18 at 16 44 20

#### Result after fix 
Was fixed by appliing next changes https://github.com/magento/magento2/pull/33849/files#diff-1d2d0f2340ff4fc3e0907e98b2ccd1a21a35c777c503907bc3c12f0223eaeb8eR254

Correct status was set, i can see **Pause**, **Cancel** , **Preview**  actions for queue

Screenshot 2021-08-18 at 16 48 58

## Issue case 2
1. After **Issue case 1** was fixed next issue was reproduced  
#### Actual result

All data were removed from Newsletter Queue after clicking **Save and Resume** button

https://user-images.githubusercontent.com/13517181/129911385-51cd3ae5-a676-4649-90f7-0176550e2f92.mov

Screenshot 2021-08-18 at 16 59 08

#### Result after fix 
Was fixed by appliing next changes  https://github.com/magento/magento2/pull/33849/files#diff-f4e00fe342edaab019fbb62abfa0c2b8f5d7a2f179bd5880e4c217d57bbb7883R232 

Data was saved correctly  all data present 

https://user-images.githubusercontent.com/13517181/129914043-f9a78778-f169-481c-b907-1b6a7dad6057.mov

#### Actual result
1. **Save and Resume** button always display  even while creating new  **Newsletter Queue**  

Screenshot 2021-08-18 at 17 04 30

#### Result after fix 
Was fixed by appliing next changes  https://github.com/magento/magento2/pull/33849/files#diff-d337328a9078b6f37102b714ffa5460b365c7ede7b97ca644bf2f0b8fe15b0dcR138

1. **Save and Resume** button displays  only if  **Newsletter Queue**  has status **Pause** 

https://user-images.githubusercontent.com/13517181/129914610-f2be7644-a867-40f8-af93-0424de400b15.mov

### Resolved issues:
1. [x] resolves magento/magento2#36612: Fixed bugs and add improvements for Newsletter Queue

### CODE REVIEW

**Comment by Usik2203** [comment]
@magento run all tests

File: app/code/Magento/Newsletter/Model/Queue.php [hunk 1]
```diff
@@ -283,6 +287,19 @@ public function sendPerSubscriber($count = 20)
...
 
+    /**
+     * Start queue: set status SENDING for queue
+     *
+     * @return $this
+     */
+    private function startQueue()
+    {
+        $this->setQueueStatus(self::STATUS_SENDING);
+        $this->save();
```
Comment (Den4ik) [inline app/code/Magento/Newsletter/Model/Queue.php:307]: Why do we still use deprecated method in core?

**Review by Den4ik — CHANGES_REQUESTED** [review]

**Comment by engcom-Charlie** [comment]
@magento give me test instance

**Comment by engcom-Charlie** [comment]
@magento give me test instance

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@engcom-Charlie Will you help with updating PR?

**Comment by engcom-Charlie** [comment]
Hi @Den4ik,

Yeah, will work on the review comments and on build failures.

Thank you!

**Comment by Den4ik** [comment]
@engcom-Charlie That's great 👍

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run Unit Tests ,Integration Tests

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests CE test

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests CE

**Comment by engcom-Charlie** [comment]
Hi @Den4ik,

Thank you for your contribution!

With respect to your review comment, currently magento have no service contract/repository implementation for this module. Hence for this PR we have to consider the provided approach only but we will surly convey your thought to the core team and the PO. But for this PR we have to go with the given approach.

Apart from that if you please go ahead and complete the review process so that we can proceed it further to Ready For Testing.

Meanwhile I am looking into the build failures.

Thank you!

**Comment by engcom-Charlie** [comment]
@magento give me test instance

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
Fixed all test failures apart from functional test failure. 

AdminEditCustomerWithAssociatedNewsletterQueueTestCest.xml is the functional test which is currently failing.
image

After analysis found that, the mentioned test is not failing because of changes made in PR or not part of the PR changes. It's related to the WYSIWYG editor.

@Den4ik, requesting you to review the PR so that we can move further for testing. Moving it to Review In Progress.

Thank you!

**Review by Den4ik — APPROVED** [review]
✅ Approved

**Comment by Den4ik** [comment]
@engcom-Charlie You may proceed with this PR.
@Usik2203 Thanks for your work

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by engcom-Lima** [comment]
:heavy_check_mark: QA Passed

**Preconditions**

1. For correct testing you should have more then 20 subscribers in newsletter_subscriber table
2. Create new Newsletter Templates in Marketing->Newsletter Templates
3. Create new Newsletter Queue

**Manual testing scenario:**
1. Run bin/magento cron:run --group=default , newsletter_send_all cron job should run Newsletter Queue
2. You will see that 20 emails were sent but we still have Not Sent status
3. Check other option available other than preview -ex pause (issue 1)
4. Pause and click queue ->Save and resume and check data available or not.
5. Save and Resume button always display even while creating new Newsletter Queue

**Before:** :heavy_multiplication_x: Mentioned issues were reproduced.
Screenshot 2022-09-07 at 5 30 02 PM

**After:** :heavy_check_mark:  Resolved all the issues.
Screenshot 2022-09-07 at 5 46 03 PM
Screenshot 2022-09-07 at 5 45 16 PM

Screenshot 2022-09-07 at 5 45 16 PM

Checks are failed. Hence, moving this PR to **Extended testing**

Thanks

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests CE, Integration Tests

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests CE

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run Integration Tests, Functional Tests EE

**Comment by engcom-Lima** [comment]
@magento create issue

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run Functional Tests CE

**Comment by engcom-Echo** [comment]
Functional Tests B2B, Functional Tests EE,Functional Tests CE Failures are related to Currency Converter API.
Hence Moving to `Merge In Progress`

**Pull request merged**